deltacat/compute/compactor/steps/dedupe.py

```diff
@@ -168,134 +174,147 @@ def dedupe(
     num_materialize_buckets: int,
     dedupe_task_index: int,
     delete_old_primary_key_index: bool,
+    enable_profiler: bool,
 ) -> DedupeResult:
 
     logger.info(f"[Dedupe task {dedupe_task_index}] Starting dedupe task...")
-    # TODO (pdames): mitigate risk of running out of memory here in cases of
-    #  severe skew of primary key updates in deltas
-    src_file_records_obj_refs = [
-        cloudpickle.loads(obj_id_pkl) for obj_id_pkl in object_ids
-    ]
-    logger.info(
-        f"[Dedupe task {dedupe_task_index}] Getting delta file envelope "
-        f"groups for {len(src_file_records_obj_refs)} object refs..."
-    )
-
-    delta_file_envelope_groups_list = ray.get(src_file_records_obj_refs)
-    hb_index_to_delta_file_envelopes_list = defaultdict(list)
-    for delta_file_envelope_groups in delta_file_envelope_groups_list:
-        for hb_idx, dfes in enumerate(delta_file_envelope_groups):
-            if dfes is not None:
-                hb_index_to_delta_file_envelopes_list[hb_idx].append(dfes)
-    src_file_id_to_row_indices = defaultdict(list)
-    deduped_tables = []
-    logger.info(
-        f"[Dedupe task {dedupe_task_index}] Running {len(hb_index_to_delta_file_envelopes_list)} "
-        f"dedupe rounds..."
-    )
-    for hb_idx, dfe_list in hb_index_to_delta_file_envelopes_list.items():
-        logger.info(f"{dedupe_task_index}: union primary keys for hb_index: {hb_idx}")
-
-        table, union_time = timed_invocation(
-            func=_union_primary_key_indices,
-            s3_bucket=compaction_artifact_s3_bucket,
-            round_completion_info=round_completion_info,
-            hash_bucket_index=hb_idx,
-            df_envelopes_list=dfe_list,
+    task_id = get_current_ray_task_id()
+    worker_id = get_current_ray_worker_id()
+    with memray.Tracker(
+        f"dedupe_{worker_id}_{task_id}.bin"
```

Comment:
Add documentation on how to analyze these files or a link to the docs that provides this info.

Reply:
Added link to documentation and track compaction round memory usage.